packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/endpoint_client/endpoint_client.dart

```diff
@@ -52,6 +109,18 @@ class EndpointClient {
   Future<void> removeMetric(String name) async =>
       _globalFieldsManager.removeMetric(name);
 
+  /// Set provided fields of the Endpoint
+  void setFields({
```

Comment:
So if I don't pass a value and it's already set, it gets overridden with `null`? That's a bit confusing. Although partly a limitation of the Dart type system, I wouldn't recommend that for an API.

Reply:
Indeed, it would be better to have different setter fields then.